src/utils/normalizeNonFinite.ts

```diff
@@ -0,0 +1,13 @@
+export function normalizeNonFinite<T>(input: T): T {
+  function walk(v: any): any {
+    if (v === 'Infinity') return Infinity;
+    if (v === '-Infinity') return -Infinity;
+    if (v === 'NaN') return NaN;
+    if (Array.isArray(v)) return v.map(walk);
+    if (v && typeof v === 'object') {
+      for (const k of Object.keys(v)) (v as any)[k] = walk((v as any)[k]);
```

Comment:
I think it would be safer to avoid in-place mutation when normalizing nested objects.

Right now, in `useProxy` we shallow-copy the event, but its `nativeEvent` still holds references to nested objects. If `normalizeNonFinite` mutates those in place, we might accidentally modify objects that other listeners or RN internals are still holding on to.

Instead, we can build a fresh object during recursion to avoids side effects and keeps the normalizer a pure transformation function:

```suggestion
      const out: any = {};
      for (const k of Object.keys(v)) out[k] = walk((v as any)[k]);
      return out;
```

Reply:
Nice catch, fixed it in 15ec8e96